### PR TITLE
Fixes on git hooks

### PR DESCRIPTION
**Pull request by ShoboySnr**
fix why git hooks was not working and added the pre-push and pre-commit hooks

### CODE REVIEW

File: package.json
Comment (dhyon) [inline package.json:13]: on this branch with on a clean install, this step fails 👀 

you may need to bump up the version of husky to resolve this

File: .husky/pre-commit
Comment (dhyon) [inline .husky/pre-commit:4]: we'd like to run lint-staged + prettier as well as a part of the pre-commit right?

File: package.json
Comment (ShoboySnr) [reply package.json:13]: Yes I just did.

File: .husky/pre-commit
Comment (ShoboySnr) [reply .husky/pre-commit:4]: Okay, will add them asap.

**Review by dhyon — APPROVED** [review]
So I've spent about an hour going through all the solutions listed here: https://stackoverflow.com/questions/50048717/lint-staged-not-running-on-precommit?answertab=votes#tab-top but I haven't gotten these hooks to trigger for me 🥲 

If this personally works for you, feel free to merge this change after adding any necessary instructions to the README for others to also benefit

**Review by muhdmud — APPROVED** [review]

**Pull request closed without merging**
